src/scripts/main.js

```diff
@@ -1,3 +1,55 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise((resolve, reject) => {
+  const logo = document.getElementsByClassName('logo')[0];
+
+  if (logo) {
+    logo.addEventListener('click', e => {
+      e.stopPropagation();
+      resolve('Promise was resolved!');
+    });
+  };
+
+  document.addEventListener('click', (e) => {
+    reject(new Error('Promise was rejected!'));
+  });
+});
+
+promise1
+  .then((result) => {
+    const newDiv = document.createElement('DIV');
+
+    newDiv.classList.add('message');
+    newDiv.innerHTML = result;
+    document.body.appendChild(newDiv);
+  })
+  .catch((error) => {
+    const newDiv = document.createElement('DIV');
+
+    newDiv.classList.add('message', 'error-message');
+    newDiv.innerHTML = error;
```

Comment:
Just a suggestion - use `textContent` instead of `innerHTML` to set the text content of the newDiv element for better security and performance.